Sources/GPX/OAGPXDocumentPrimitives.h

```diff
@@ -81,97 +145,73 @@ struct RouteDataBundle;
 
 @end
 
-@interface OAMetadata : NSObject
+@interface OAMetadata : OAGpxExtensions
 
 @property (nonatomic) NSString *name;
 @property (nonatomic) NSString *desc;
 @property (nonatomic) NSArray *links;
 @property (nonatomic) long time;
-@property (nonatomic) OAExtraData *extraData;
 
 @end
 
-@interface OALocationMark : NSObject<OALocationPoint>
+@interface OAWptPt : OAGpxExtensions<OALocationPoint>
+
+@property (nonatomic, assign) std::shared_ptr<OsmAnd::GpxDocument::GpxWptPt> wpt;
 
 @property (nonatomic) BOOL firstPoint;
 @property (nonatomic) BOOL lastPoint;
 @property (nonatomic) CLLocationCoordinate2D position;
 @property (nonatomic) NSString *name;
 @property (nonatomic) NSString *desc;
-@property (nonatomic) CLLocationDistance elevation;
+@property (nonatomic) double elevation;
```

Comment:
Very wrong! CLLocationDistance IS a double, no need for this change as it makes the code less readable.